python/plugins/processing/algs/gdal/AssignProjection.py

```diff
@@ -94,6 +112,24 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
 
         commands.append(GdalUtils.escapeAndJoin(arguments))
 
+        # if the input layer is a project's layer, the algorithm can
+        # update the QgsMapLayer CRS
+        if context.project() and context.project().mapLayer(inLayer.id()):
```

Comment:
This isn't safe to call from a background thread -- you'd need to make the algorithm run in the main thread to be able to access QgsProject.mapLayer()

Reply:
I don't want to run the algorithm in the main thread. I would like to be sure that the input which is the output will be updated at the end of the algorithm.

* How to know that the input layer is already loaded in the project context ?
* If the algorithm is run in an other thread than the main one, is the `context.project()` not null even if it is not safe to access `mapLayer()` ? 
* How to reload or update the input QgsRasterLayer, that is also the output, at the end of the algorithm ? I try to solve https://github.com/qgis/QGIS/issues/37920

Reply:
I'd suggest overriding the postProcessAlgorithm method (which always runs in the main thread) and do the update there.

Reply:
thanks Nyall, I will try

Reply:
Sorry I missed off the rest of my advice! I was going to suggest that the post processor runs through all the layers in the project, and updates all of them where the layers source matches the changed layer's source. Then it'll nicely handle the case that the layer is loaded multiple times in the prpject

Reply:
Thanks @nyalldawson, I have overrided postProcessAlgorithm method and I was able to get the QgsRasterLayer's crs updated in the project and the processing context.